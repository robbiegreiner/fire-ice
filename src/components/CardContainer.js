import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Card from './Card.js';

class CardContainer extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div>
        <h1>hi</h1>
      </div>
    );
  }
}

CardContainer.propTypes = {
  houseData: PropTypes.object
};

export default CardContainer;
