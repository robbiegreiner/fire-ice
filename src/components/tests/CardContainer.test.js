import React from "react";
import CardContainer from '../CardContainer';
import { shallow } from "enzyme";

describe('CardContainer component', () => {
  const mockFn = jest.fn();

  const wrapper = shallow(<CardContainer
    store
    houseData={[{
      name: 'sure',
      founded: 'why not',
      seats: ['ok', 'cool'],
      titles: ['ok', 'cool'],
      coatOfArms: 'cool dude',
      ancestralWeapons: ['ok', 'cool'],
      words: ['whatever']
    }]}/>);


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
