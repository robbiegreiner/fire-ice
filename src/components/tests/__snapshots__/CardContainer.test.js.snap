// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    houseData={
        Array [
            Object {
              "ancestralWeapons": Array [
                "ok",
                "cool",
              ],
              "coatOfArms": "cool dude",
              "founded": "why not",
              "name": "sure",
              "seats": Array [
                "ok",
                "cool",
              ],
              "titles": Array [
                "ok",
                "cool",
              ],
              "words": Array [
                "whatever",
              ],
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          house={
                    Object {
                              "ancestralWeapons": Array [
                                "ok",
                                "cool",
                              ],
                              "coatOfArms": "cool dude",
                              "founded": "why not",
                              "name": "sure",
                              "seats": Array [
                                "ok",
                                "cool",
                              ],
                              "titles": Array [
                                "ok",
                                "cool",
                              ],
                              "words": Array [
                                "whatever",
                              ],
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "sure",
        "nodeType": "function",
        "props": Object {
          "house": Object {
            "ancestralWeapons": Array [
              "ok",
              "cool",
            ],
            "coatOfArms": "cool dude",
            "founded": "why not",
            "name": "sure",
            "seats": Array [
              "ok",
              "cool",
            ],
            "titles": Array [
              "ok",
              "cool",
            ],
            "words": Array [
              "whatever",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "ok",
                                      "cool",
                                    ],
                                    "coatOfArms": "cool dude",
                                    "founded": "why not",
                                    "name": "sure",
                                    "seats": Array [
                                      "ok",
                                      "cool",
                                    ],
                                    "titles": Array [
                                      "ok",
                                      "cool",
                                    ],
                                    "words": Array [
                                      "whatever",
                                    ],
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "sure",
          "nodeType": "function",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "ok",
                "cool",
              ],
              "coatOfArms": "cool dude",
              "founded": "why not",
              "name": "sure",
              "seats": Array [
                "ok",
                "cool",
              ],
              "titles": Array [
                "ok",
                "cool",
              ],
              "words": Array [
                "whatever",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
